src/mongo_types.ts

```diff
@@ -541,3 +525,60 @@ export type NestedPathsOfType<TSchema, Type> = KeysOfAType<
   },
   Type
 >;
+
+/**
+ * @public
+ * @experimental
+ */
+export type DotNotationFilter<TSchema> =
```

Comment:
We should throw around some more ideas for a new name here, if we want to iron this type out more and eventually reintroduce it I wouldn't want folks to go through another rename change, is there something more forward looking? StrictFilter? SmartFilter? MongoDBFilter? We can thread in slack and resolve this comment with team consensus

Reply:
Well, I think that regardless of the name of this type, the actual filter type should (and will always) be named `Filter` because that aligns with every piece of documentation I've ever seen about mongodb.

This is intended to be a temporary workaround, I don't think a rename change is a big deal.

If down the line, we decide we don't want to force users to rename, we can always just alias the current name to the `Filter` type:
```typescript
export type DotNotationFilter<T> = Filter<T>;
```

Reply:
> actual filter type should ... be named `Filter`

Yes I agree

> that aligns with every piece of documentation I've ever seen about mongodb

It used to be named FilterQuery in 3.x community types, and query has been interchangeable with filter, but we unified it all across the driver in 4.0.

> we can always just alias the current name to the Filter

Yep although we wouldn't want to maintain a duplicate type forever. I think if we consider a more generic name the type can prove useful as an opt-in TS feature for folks that has more than dot-notation.

I am thinking about how typescript offers increasing safety with each version sometimes introducing new behavior that require code changes. The approach taken by TS is to put these changes behind the strict flag, non-strict TS does not break between compiler versions (to my knowledge). Along that line of thinking, I'd like us to consider `StrictFilter` as this annotation does provide assertions on nested types. In the future, we could consider enhancements that increase the hinting & assertions to improve DX, possibly [taking ideas from here](https://docs.google.com/document/d/1IwjKAiBUXLm9WlpJdw6eSgGPdGp4i-a809vsVK3S6FI/edit#heading=h.5qau76drh2m1).